src/ScheduleFunctions.cpp

```diff
@@ -1381,12 +1381,21 @@ class InjectFunctionRealization : public IRMutator {
                                                                   replacements, add_lets);
                 producer = inject_stmt(producer, produceDef, f.definition().schedule().fuse_level().level);
             }
+        }
+
+        bool some_updated = false;
+        for (size_t j = 0; j == 0 || some_updated; j++) {
```

Comment:
Minor suggestions: Why not just initialize some_updated to true and simplify the loop condition?